configuration/configuration.go

```diff
@@ -466,31 +467,51 @@ func (c *ConfigurationData) GetCacheControlUser() string {
 	return c.v.GetString(varCacheControlUser)
 }
 
+// use a mutex to prevent concurrent parsing
+var privateKeyParsingMutex sync.Mutex
+
 // GetTokenPrivateKey returns the private key (as set via config file or environment variable)
 // that is used to sign the authentication token.
 func (c *ConfigurationData) GetTokenPrivateKey() (*rsa.PrivateKey, error) {
 	if c.tokenPrivateKey == nil {
 		if !c.v.IsSet(varTokenPrivateKey) {
 			return nil, errors.Errorf("'token.privatekey' variable is not defined")
 		}
-		var err error
-		c.tokenPrivateKey, err = jwt.ParseRSAPrivateKeyFromPEM([]byte(c.v.GetString(varTokenPrivateKey)))
-		return c.tokenPrivateKey, err
+
+		// lock the mutex before parsing the key, so if another rountine is already parsing the key, then this one waits
+		privateKeyParsingMutex.Lock()
+		defer privateKeyParsingMutex.Unlock()
```

Comment:
We must always lock the mutex, so these two lines need to be the first one in this function.

Reply:
why always lock the mutex ? there's no need for that if the key was already parsed, isn'it ?

Reply:
got it now